src/mergeTests.ts

```diff
@@ -336,23 +370,165 @@ function insertAnnotations(
+
+      addCounter -= 1;
+    } else {
+      locationCounter -= 1;
+    }
+  }
+
+  // Insert any remaining imports after package statement
+  while (addCounter > -1) {
+    const importToAdd = allImports[addCounter];
+    const [newlinesAfter, newlinesBefore] = pickNewlines(pkgLocation + 1, importLocations, `import ${importToAdd}`, pkgLocation);
+    spliceCode(pkgLocation + 1, splitCode, `import ${importToAdd};`, newlinesBefore, newlinesAfter);
+
+    importLocations.splice(0, 0, [importToAdd, pkgLocation + 1]);
+
+    addCounter--;
+  }
+}
+
+/**
+ * Count newlines
+ *
+ * @param {string} input
+ * @returns
+ */
+function nlc(input: string) {
+  const matches = input.match(/^(\n+)((.|$)+$)/);
+  return matches ? matches[1].length : 0;
+}
+
+/**
+ * Leave string with only one preceeding newline
+ *
+ * @param {string} input string
+ * @returns
+ */
+function oneNewline(input: string) {
+  // leaves one newline in place
+  const split = input.match(/^(\n+)((.|\n|$)+)/);
+  return split ? `\n${split[2]}` : `\n${input}`;
+}
+
+/**
+ * determine whether an import should be surrounded by newliens
+ *
+ * @param {number} loc  location of import
+ * @param {string[]} splitCode splitCode
+ * @param {string} importToAdd  import being added
+ * @returns
+ */
+function pickNewlines(loc: number, importLocations: Array<[string, number]>, importToAdd: string, pkg?: number) {
+  let newlineAfter = 0;
+  let newlineBefore = 0;
+  const addition = getImportGroup(importToAdd.trim().split(' ')[1]);
+  // the line after the current import
+  // const afterImport = splitCode[loc].trim().split(' ')[1];
+  const afterImport = importLocations.filter((i: [string, number]) => i[1] === loc);
+  let afterImportGroup;
+  if (afterImport[0] && afterImport[0][0]) {
+    afterImportGroup = getImportGroup(afterImport[0] && afterImport[0][0]);
+  }
+  // the line before the current import
+  // const beforeImport = splitCode[loc - 1] && splitCode[loc - 1].trim().split(' ')[1];
+  const beforeImport = importLocations.filter((i: [string, number]) => i[1] === loc - 1);
+  const beforeImportGroup = getImportGroup(beforeImport[0] && beforeImport[0][0]);
+
+  // if import types are dissimilar, there must be a blank line between them
+  if (addition !== beforeImportGroup) {
+    newlineBefore = 1;
+  }
+
+  if (beforeImport.length === 0 && pkg === -1) {
+    // we're at the beginning, there's no package
+    newlineBefore = -1;
+  }
+
+  // a space should be added if the next line isn't an import
+  if (addition !== afterImportGroup) {
+    newlineAfter = 1;
+  }
+
+  // same type no space
+  if (addition === afterImportGroup) {
+    newlineAfter = -1;
+  }
 
-  if (importLocations.length && importLocations[importLocations.length - 1]) {
-    importInsertLocation = importLocations[importLocations.length - 1][1] + 1;
+  // a newline should be added if we're at package
+  if ((loc - 1 === pkg) && pkg !== -1) {
+    newlineBefore = 1;
+  }
+  return [newlineAfter, newlineBefore];
```

Comment:
"after, before"??? Please fix (and fix in the places this is called)

```suggestion
  return [newlineBefore, newlineAfter];
```